### PR TITLE
Invalid Authentication Link Alert

### LINKED ISSUE #1605: Magic Login Authentication
### Details
- [x] Integrate the new Login Endpoints
- [x] Implement a new UI that would allow users to log in using their emails
- [x] Display an alert whenever we successfully request an Auth Link
- [x] Display an alert whenever the user interacts with a link that's no longer valid

### PR DESCRIPTION
**Pull request by jleandroperez**
### Fix
In this PR we're building a UI, to be presented whenever the user attempts to authenticate using a link that's no longer valid.

Closes #1605

### Test: From Onboarding
1. Fresh install Simplenote iOS
2. Make sure you're logged out!
3. Switch to Safari and open: `simplenote://login?auth_key=1234&auth_code=5678`

- [x] Verify Simplenote ends up in the Foreground
- [x] Verify that shortly after you get an alert, indicating that the Magic Link isn't valid
- [x] Verify that pressing the `Request new Link` button gets you back into the Login UI

### Test: From Login UI!
1. Fresh install Simplenote iOS
2. Make sure you're logged out!
3. Press on the `Log In` button
4. Switch to Safari and open: `simplenote://login?auth_key=1234&auth_code=5678`

- [x] Verify that the Invalid Link Alert is also presented
- [x] Verify that pressing the `Request new Link` button gets you back into the Login UI

### Test: From Login UI!
1. Fresh install Simplenote iOS
2. Make sure you're logged out!
3. Press on the `Sign Up` button
4. Switch to Safari and open: `simplenote://login?auth_key=1234&auth_code=5678`

- [x] Verify that the Invalid Link Alert is also presented
- [x] Verify that pressing the `Request new Link` button gets you back into the Login UI

### Release
These changes do not require release notes.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Simplenote/MagicLinkInvalidView.swift [hunk 1]
```diff
@@ -0,0 +1,92 @@
+import Foundation
+import SwiftUI
+import Gridicons
+
+
+// MARK: - MagicLinkConfirmationView
+//
+struct MagicLinkInvalidView: View {
```
Comment (charliescheer) [inline Simplenote/MagicLinkInvalidView.swift:8]: I think there is some duplication here with the invalid view and the check your email view.  They are the same except that the invalid view has the new link button.  Not sure if it is worth it, but I wonder if there is a way to save on the duplication here

**Review by charliescheer — APPROVED** [review]
👍

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by jleandroperez** [comment]
RE: MagicLinkView(s), we'll need to revisit the UX for this flow. Following up in another PR.

Thank you Charlie!!

**Pull request merged**